### PR TITLE
xcodegen 1.10.3 (new formula)

### PR DESCRIPTION
**Pull request by yonaskolb**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by ilovezfs** [comment]
@yonaskolb the test block is failing on 10.13 with

```
==> brew test xcodegen --verbose
==> FAILED
Testing xcodegen
/usr/bin/sandbox-exec -f /private/tmp/homebrew20180714-33892-okopvu.sb /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew:/usr/local/Homebrew/Library/Homebrew/cask/lib -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/xcodegen.rb --verbose
==> /usr/local/Cellar/xcodegen/1.10.3/bin/XcodeGen --spec /private/tmp/xcodegen-test-20180714-33893-1tj01kt/xcodegen.yml
📋  Loaded project:
  Name: GeneratedProject
  Targets:
    📱  TestProject: application
⚙️  Generating project...
⚙️  Writing project...
💾  Saved project to /tmp/xcodegen-test-20180714-33893-1tj01kt/GeneratedProject.xcodeproj
==> xcodebuild -project /private/tmp/xcodegen-test-20180714-33893-1tj01kt/GeneratedProject.xcodeproj
=== BUILD TARGET TestProject OF PROJECT GeneratedProject WITH THE DEFAULT CONFIGURATION () ===

Check dependencies
Code Signing Error: Bundle identifier is missing. TestProject doesn't have a bundle identifier for the Debug build configuration. Add a value for PRODUCT_BUNDLE_IDENTIFIER in the build settings editor.
Code Signing Error: Code signing is required for product type 'Application' in SDK 'iOS 11.4'

** BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
Error: xcodegen: failed
Failed executing: xcodebuild -project /private/tmp/xcodegen-test-20180714-33893-1tj01kt/GeneratedProject.xcodeproj
/usr/local/Homebrew/Library/Homebrew/formula.rb:1856:in `block in system'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1794:in `open'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1794:in `system'
/usr/local/Homebrew/Library/Homebrew/extend/fileutils.rb:59:in `xcodebuild'
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/xcodegen.rb:34:in `block in <class:Xcodegen>'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1700:in `block (3 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/utils.rb:557:in `with_env'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1699:in `block (2 levels) in run_test'
/usr/local/Homebrew/Library/Homebrew/formula.rb:857:in `with_logging'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1698:in `block in run_test'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1906:in `block in mktemp'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:55:in `block in run'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:55:in `chdir'
/usr/local/Homebrew/Library/Homebrew/mktemp.rb:55:in `run'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1905:in `mktemp'
/usr/local/Homebrew/Library/Homebrew/formula.rb:1692:in `run_test'
/usr/local/Homebrew/Library/Homebrew/test.rb:28:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/timeout.rb:91:in `block in timeout'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/timeout.rb:33:in `block in catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/timeout.rb:33:in `catch'
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/timeout.rb:106:in `timeout'
/usr/local/Homebrew/Library/Homebrew/test.rb:27:in `<main>'
```

File: Formula/xcodegen.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      options:
+        bundleIdPrefix: com.project
+      targets:
+        TestProject:
+          type: application
+          platform: iOS
+          sources: TestProject
+    EOS
+    (testpath/"TestProject").mkpath
+    system bin/"XcodeGen", "--spec", testpath/"xcodegen.yml"
```
Comment (ilovezfs) [inline Formula/xcodegen.rb]: Can we assert something about what exists and the content of what exists after this step? See `mkcert` for instance:
https://github.com/Homebrew/homebrew-core/blob/master/Formula/mkcert.rb#L29-L34

Comment (yonaskolb) [reply Formula/xcodegen.rb]: Is that necessary? Xcodegen will fail if reading, parsing, generating or writing fails. The fact that it got through all those things is a valid test in my opinion. 
Validating the actuall structure of the project should be left to the actual xcodegen tests

Comment (ilovezfs) [reply Formula/xcodegen.rb]: yes, it should assert something other than just 0 exit code

Comment (yonaskolb) [reply Formula/xcodegen.rb]: Ok, I'll see if I can add something. Is there a way to run the tests locally? I couldn't find the command for that

Comment (ilovezfs) [reply Formula/xcodegen.rb]: `brew test --verbose --debug xcodegen`

Comment (yonaskolb) [reply Formula/xcodegen.rb]: Yeah I tried that but get a `Error: xcodegen is not linked` error

Comment (ilovezfs) [reply Formula/xcodegen.rb]:
```
brew link xcodegen
```

**Comment by ilovezfs** [comment]
Thanks for the new formula @yonaskolb! You're awesome.

**Comment by yonaskolb** [comment]
Thanks for the merge @ilovezfs!

**Pull request closed without merging**